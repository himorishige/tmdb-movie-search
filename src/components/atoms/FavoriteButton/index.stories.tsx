import { Meta, Story } from '@storybook/react';
import FavoriteButton, { Props } from '.';

export default {
  title: 'TMDB Search/Atoms/FavoriteButton',
  component: FavoriteButton,
} as Meta;

const Template: Story<Props> = (args) => <FavoriteButton {...args} />;

export const Default = Template.bind({});
Default.args = {
  favoriteFlag: true,
  size: '2rem',
  toggleHandler: () => {},
};
Default.decorators = [
  (StoryFn) => <div style={{ padding: '2rem', backgroundColor: '#2A4365' }}>{StoryFn()}</div>,
];
