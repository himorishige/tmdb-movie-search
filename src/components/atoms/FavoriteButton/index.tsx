import { memo } from 'react';
import { IconButton } from '@chakra-ui/react';
import { MdFavorite } from 'react-icons/md';

export type Props = {
  favoriteFlag: boolean;
  size?: string;
  toggleHandler: () => void;
};

const FavoriteButton: React.VFC<Props> = memo((props) => {
  const { favoriteFlag, toggleHandler, size = '1rem' } = props;

  return (
    <>
      {favoriteFlag ? (
        // お気に入りに登録済みの場合
        <IconButton
          type="button"
          aria-label="add FavoriteList"
          icon={<MdFavorite fontSize={size} />}
          color="red.400"
          _hover={{
            backgroundColor: 'transparent',
            color: { base: 'red.400', md: 'gray.100' },
            transform: 'scale(1.3)',
          }}
          transition="all 0.3s"
          bgColor="transparent"
          onClick={toggleHandler}
          _focus={{ outline: 'none', background: 'none' }}
          _active={{ outline: 'none', background: 'none' }}
          data-testid="remove-favorites-button"
        />
      ) : (
        // お気に入りに未登録の場合
        <IconButton
          type="button"
          aria-label="remove FavoriteList"
          icon={<MdFavorite fontSize={size} />}
          color="gray.100"
          _hover={{
            backgroundColor: 'transparent',
            color: { base: 'gray.100', md: 'red.400' },
            transform: 'scale(1.3)',
          }}
          transition="color 0.3s"
          bgColor="transparent"
          onClick={toggleHandler}
          _focus={{ outline: 'none', background: 'none' }}
          _active={{ outline: 'none', background: 'none' }}
          data-testid="add-favorites-button"
        />
      )}
    </>
  );
});

export default FavoriteButton;
