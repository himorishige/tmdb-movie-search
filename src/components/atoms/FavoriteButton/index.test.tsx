import renderer from 'react-test-renderer';
import FavoriteButton, { Props } from '.';

describe('Atoms/FavoriteButton', () => {
  test('正しくレンダリングされている', () => {
    const props: Props = {
      favoriteFlag: true,
      toggleHandler: jest.fn(),
    };
    const button = renderer.create(<FavoriteButton {...props} />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
