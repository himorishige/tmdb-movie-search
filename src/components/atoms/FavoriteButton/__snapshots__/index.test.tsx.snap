// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/FavoriteButton 正しくレンダリングされている 1`] = `
<button
  aria-label="add FavoriteList"
  className="chakra-button css-1ijdlh6"
  data-testid="remove-favorites-button"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    aria-hidden={true}
    fill="currentColor"
    focusable={false}
    fontSize="1rem"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
    />
  </svg>
</button>
`;
